lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -401,8 +402,11 @@ withNetworkLayer tr network np project k = do
         BF.Block {_blockSlot} <- bfGetLatestBlock bfLayer
         let latestSlot = maybe 0 BF.unSlot _blockSlot
             currentSlot = fromIntegral (unSlotNo s)
-            percentage = currentSlot % latestSlot
-        pure $ Syncing $ Quantity $ clipToPercentage percentage
+            percentage = clipToPercentage $ currentSlot % latestSlot
+        pure $
+            if percentageToDouble percentage > 0.99
```

Comment:
By the way, for numerical algorithms, there is a rule of thumb which says "Never compare `Double`!", as this usually leads to numerical instability and weird artifacts. (For example, GHCi will evaluate `3*0.1 > 0.3` to `True`.)

The less extreme variant of this rule is to "Always double check comparison of `Double`". I have double-checked this code and I believe that it is perfectly fine here. 🤓

Reply:
Yeah, I think we're good in this particular case as there is no much arithmetic to loose precision and we can tolerate some imprecision as there is no `==` operator in use. 
https://www.geeksforgeeks.org/problem-in-comparing-floating-point-numbers-and-how-to-compare-them-correctly/